target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr1043n-v5.c

```diff
@@ -0,0 +1,194 @@
+
+#define TL_WR1043N_V5_GPIO_BTN_RESET            2
+#define TL_WR1043N_V5_GPIO_BTN_WPS_WIFI         5
+#define TL_WR1043N_V5_KEYS_POLL_INTERVAL        20     /* msecs */
+#define TL_WR1043N_V5_KEYS_DEBOUNCE_INTERVAL    (3 * TL_WR1043N_V5_KEYS_POLL_INTERVAL)
+
+#define TL_WR1043N_V5_MAC0_OFFSET               0
+#define TL_WR1043N_V5_MAC1_OFFSET               6
+#define TL_WR1043N_V5_WMAC_CALDATA_OFFSET       0x1000
+
+#define TL_WR1043N_V5_GPIO_MDC			3
+#define TL_WR1043N_V5_GPIO_MDIO			4
+
+extern void gpio_set_value(unsigned gpio, int value);
+
+
+static struct gpio_led tl_wr1043n_v5_leds_gpio[] __initdata = {
+	{
+		.name		= "tp-link:green:wlan",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_WLAN,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:system",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_STATUS,
+		.active_low	= 1,
+		.default_state = LEDS_GPIO_DEFSTATE_ON,
+	},
+	{
+		.name		= "tp-link:green:lan_1",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_LAN1,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:lan_2",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_LAN2,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:lan_3",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_LAN3,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:lan_4",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_LAN4,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:orange:wan_link",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_WAN_LINK,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:wan_inet",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_WAN_INET,
+		.active_low	= 1,
+	},
+	{
+		.name		= "tp-link:green:wps",
+		.gpio		= TL_WR1043N_V5_GPIO_LED_WPS,
+		.active_low	= 1,
+	},
+};
+
+static struct gpio_keys_button tl_wr1043n_v5_gpio_keys[] __initdata = {
+        {
+                .desc           = "WPS and WIFI button",
+                .type           = EV_KEY,
+                .code           = KEY_WPS_BUTTON,
+                .debounce_interval = TL_WR1043N_V5_KEYS_DEBOUNCE_INTERVAL,
+                .gpio           = TL_WR1043N_V5_GPIO_BTN_WPS_WIFI,
+                .active_low     = 1,
+        },
+        {
+                .desc           = "Reset button",
+                .type           = EV_KEY,
+                .code           = KEY_RESTART,
+                .debounce_interval = TL_WR1043N_V5_KEYS_DEBOUNCE_INTERVAL,
+                .gpio           = TL_WR1043N_V5_GPIO_BTN_RESET,
+                .active_low     = 1,
+        },
+};
+
+static struct ar8327_pad_cfg tl_wr1043n_v5_ar8337_pad0_cfg = {
+	.mode = AR8327_PAD_MAC_SGMII,
+	.sgmii_delay_en = true,
+};
+
+static struct ar8327_platform_data tl_wr1043n_v5_ar8337_data = {
+	.pad0_cfg = &tl_wr1043n_v5_ar8337_pad0_cfg,
+	.port0_cfg = {
+		.force_link = 1,
+		.speed = AR8327_PORT_SPEED_1000,
+		.duplex = 1,
+		.txpause = 1,
+		.rxpause = 1,
+	},
+};
+
+static struct mdio_board_info tl_wr1043n_v5_mdio0_info[] = {
+	{
+		.bus_id = "ag71xx-mdio.0",
+		.phy_addr = 0,
+		.platform_data = &tl_wr1043n_v5_ar8337_data,
+	},
+};
+
+static void __init tl_wr1043nv5_setup(void)
```

Comment:
several indentation problems in that function, please review.

Reply:
I admit it was a splice job between the v4 file and the TP-Link GPL sources.  I hope to have time to clean up the code in the next few days.  Thank you for notes.